azurerm/resource_arm_log_profile.go

```diff
@@ -0,0 +1,259 @@
+	err = resource.Retry(300*time.Second, retryLogProfilesClientGet(name, meta))
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmLogProfileRead(d, meta)
+}
+
+func resourceArmLogProfileRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logProfilesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name, err := parseLogProfileNameFromID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Log Profile %q: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	if props := resp.LogProfileProperties; props != nil {
+		d.Set("storage_account_id", props.StorageAccountID)
+		d.Set("service_bus_rule_id", props.ServiceBusRuleID)
+
+		locations := make([]string, len(*props.Locations))
+		for index, location := range *props.Locations {
+			locations[index] = azureRMNormalizeLocation(location)
+		}
+		d.Set("locations", locations)
+		d.Set("categories", props.Categories)
+
+		d.Set("retention_policy", flattenAzureRmLogProfileRetentionPolicy(props.RetentionPolicy))
+	}
+
+	return nil
+}
+
+func resourceArmLogProfileDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logProfilesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+
+	_, err := client.Delete(ctx, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Log Profile %q: %+v", name, err)
+	}
+
+	return nil
+}
+
+func expandLogProfileCategories(d *schema.ResourceData) []string {
+	logProfileCategories := d.Get("categories").([]interface{})
+	categories := []string{}
+
+	for _, category := range logProfileCategories {
+		categories = append(categories, category.(string))
+	}
+
+	return categories
+}
+
+func expandLogProfileLocations(d *schema.ResourceData) []string {
+	logProfileLocations := d.Get("locations").([]interface{})
+	locations := []string{}
+
+	for _, location := range logProfileLocations {
+		locations = append(locations, location.(string))
+	}
+
+	return locations
+}
+
+func expandAzureRmLogProfileRetentionPolicy(d *schema.ResourceData) (*insights.RetentionPolicy, error) {
+	retentionPolicies := d.Get("retention_policy").([]interface{})
+
+	if len(retentionPolicies) > 0 {
+		retentionPolicy := retentionPolicies[0].(map[string]interface{})
+		logProfileRetentionPolicy := &insights.RetentionPolicy{
+			Enabled: utils.Bool(retentionPolicy["enabled"].(bool)),
+			Days:    utils.Int32(int32(retentionPolicy["days"].(int))),
+		}
+
+		return logProfileRetentionPolicy, nil
+	}
+
+	return nil, fmt.Errorf("[ERROR] Retention policy must be set")
+}
+
+func flattenAzureRmLogProfileRetentionPolicy(input *insights.RetentionPolicy) []interface{} {
+	result := make(map[string]interface{})
```

Comment:
There are several potential `nil` pointer crash possibilities here: `input`, `input.Enabled` and `input.Days`.